indexing-service/src/test/java/org/apache/druid/indexing/firehose/IngestSegmentFirehoseFactoryTimelineTest.java

```diff
@@ -286,13 +331,26 @@ private static InputRow ir(String timeString, long metricValue)
             throw new IllegalArgumentException("WTF");
           }
         }
+
+        @Override
+        public DataSegment getDatabaseSegmentDataSourceSegment(String dataSource, String segmentId)
+        {
+          return testCase.segments
+              .stream()
+              .filter(s -> s.getId().toString().equals(segmentId))
+              .findAny()
+              .get();  // throwing if not found is exactly what the real code does
```

Comment:
TeamCity doesn't want me to call `Optional.get()` without `ifPresent()` because it might throw, but I'm OK with it throwing. Not sure if I should change anything?